src/middleware.ts

```diff
@@ -10,15 +10,15 @@ import { hasApiKey } from "@/app/lib/middleware/auth";
   Consider fully validating user api key against postgres pending prisma
   client postgres support on edge runtime.
 */
-export function middleware(request: NextRequest) {
+export async function middleware(request: NextRequest) {
   const path = request.nextUrl.pathname;
   if (path.startsWith("/api/v1") && !path.startsWith("/api/v1/spec")) {
-    const authResponse = hasApiKey(request);
+    const authResponse = await validateBearerToken(request);
```

Comment:
do we need to validate the token on every request? or this is API key verification?

Reply:
we will need to validate the token on every request, since it can potentially expire at any point. however, keep in mind this is stateless (i.e. we can validate the token on the edge without accessing our db). 

did you have something else in mind for validating the token?